blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -1,39 +1,22 @@
-/*
-  ==============================================================================
-
-    blueprint_ReactApplicationRoot.h
-    Created: 9 Dec 2018 10:20:37am
-
-  ==============================================================================
-*/
-
 #pragma once
 
-#include "blueprint_EcmascriptEngine.h"
-#include "blueprint_FileWatcher.h"
-#include "blueprint_View.h"
-#include "blueprint_ViewManager.h"
-
-
 namespace blueprint
 {
-
-    //==============================================================================
     /** The ReactApplicationRoot class is the primary point of coordination between
-     *  the React.js reconciler and the native View heirarchy.
-     *
-     *  By default, ReactApplicationRoot implements a generic error handler which will
-     *  catch errors from JavaScript code and display an error screen with a stack trace
-     *  and error message. This generic error handler is enabled by default in Debug builds,
-     *  and in Release builds such errors will be thrown, with the intention that the end
-     *  user should catch and handle them appropriately.
-     *
-     *  ReactApplicationRoot also provides debug functionality similar to React Native.
-     *  Users can hit CTRL-D/CMD-D when the ReactApplicationRoot component has focus,
-     *  causing the application to suspend execution and await connection from a debug client.
-     *  See the documentation for details on setting up and connecting a debugger.
-     */
-    class ReactApplicationRoot : public View
+        the React.js reconciler and the native View heirarchy.
+
+        By default, ReactApplicationRoot implements a generic error handler which will
+        catch errors from JavaScript code and display an error screen with a stack trace
+        and error message. This generic error handler is enabled by default in Debug builds,
+        and in Release builds such errors will be thrown, with the intention that the end
+        user should catch and handle them appropriately.
+
+        ReactApplicationRoot also provides debug functionality similar to React Native.
+        Users can hit CTRL-D/CMD-D when the ReactApplicationRoot component has focus,
+        causing the application to suspend execution and await connection from a debug client.
+        See the documentation for details on setting up and connecting a debugger.
+    */
+    class ReactApplicationRoot final : public View
```

Comment:
I've also gone and marked _nearly_ everything public with `final`. I'm rather doubtful of the _necessity_ for users to override these super complex classes; it would really be a special case worth discussing imho.